spec/features/edit_dataset_spec.rb

```diff
@@ -81,21 +81,23 @@
     end
 
     it "should be able to add a new file" do
+      link = published_dataset.links.first
+
       click_change(:datalinks)
-      expect(page).to have_content("my published test file")
-      click_link 'Add another link'
+      expect(page).to have_content(link.name)
 
+      click_link 'Add another link'
       fill_in 'link[url]', with: 'http://google.com'
       fill_in 'link[name]', with: 'my other test file'
-
       click_button 'Save and continue'
-
       expect(page).to have_content('my other test file')
     end
 
     it "should be able to edit an existing file" do
+      link = published_dataset.links.first
+
       click_change(:datalinks)
-      expect(page).to have_content("my published test file")
```

Comment:
I removed all hardcoded titles/names so we can have more flexibility to change our factories without breaking the tests.